app/controllers/concerns/session_person_creator.rb

```diff
@@ -33,8 +33,12 @@ def render_or_redirect_login(person)
   private
 
     def find_or_create_person(email, &_on_create)
-      person = Person.where(email: email.to_s).first_or_initialize
-      yield(person) if person.new_record?
+      if ExternalUser.where(email: email.to_s).count.positive?
+        person = ExternalUser.where(email: email.to_s).first
```

Comment:
You should be able to use:
```suggestion
      if ExternalUser.exists?(email:)
        person = ExternalUser.find_by(email:)
```

Reply:
Updated but  .to_s was still required